src/metabase/search/ingestion.clj

```diff
@@ -115,25 +115,45 @@
   false)
 
 (defn update!
-  "Update all active engines' existing indexes with the given documents"
-  [documents-reducible]
-  (when-let [engines (seq (search.engine/active-engines))]
-    (if (= 1 (count engines))
-      (search.engine/update! (first engines) documents-reducible)
-      ;; TODO um, multiplexing over the reducible awkwardly feels strange. We at least use a magic number for now.
-      (doseq [batch (eduction (partition-all 150) documents-reducible)
-              e     engines]
-        (search.engine/update! e batch)))))
+  "Update all active engines' existing indexes with the given documents. Passed remove-documents will be deleted from the index."
+  [documents-reducible removed-models-reducible]
+  (doseq [e (seq (search.engine/active-engines))]
+    (u/prog1
+      (search.engine/update! e documents-reducible)
```

Comment:
There's a regression here (or at least there would be) in the multi-engine case.

Even if the engines themselves partition the reducible, it's important that we loop across all the engines for each batch before we proceed to the next batch, otherwise there will be head holding. Does that make sense? Since the importance of this ordering wasn't clear before, I think it deserves a comment for the future.